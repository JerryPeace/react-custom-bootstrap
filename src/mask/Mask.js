import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classNames from 'classnames';
import Loader from '../loader/Loader';

class Mask extends Component {
    static propTypes = {
        show: PropTypes.bool,
        isLoading: PropTypes.bool,
        loaderProps: PropTypes.object
    };

    static defaultProps = {
        show: true,
        loaderProps: {
            bsSize: 'md'
        }
    };

    render() {
        if (this.props.show !== true) {
            return null;
        }

        const {
            className,
            children,
            isLoading,
            loaderProps
        } = this.props;

        return (
            <div className={classNames('tm-mask', className)}>
                <div>
                    {isLoading && <Loader {...loaderProps} />}
                    {children}
                </div>
            </div>
        );
    }
}

export default Mask;
