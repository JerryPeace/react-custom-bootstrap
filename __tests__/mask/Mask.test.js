import React from 'react';
import { mount } from 'enzyme';
import toJSON from'enzyme-to-json';
import Mask from 'mask/Mask';

describe('Mask', () => {
    test('render Mask correctly', () => {
        const mask = mount(<Mask isLoading={true}/>);
        expect(toJSON(mask)).toMatchSnapshot();
    });

    test('render Mask without show correctly', () => {
        const mask = mount(<Mask show={false}/>);
        expect(toJSON(mask)).toMatchSnapshot();
    });

    test('render Mask without isLoading correctly', () => {
        const mask = mount(<Mask isLoading={false}/>);
        expect(toJSON(mask)).toMatchSnapshot();
    });

    test('render Mask with loaderProps correctly', () => {
        const mask = mount(<Mask isLoading={true} loaderProps={{ className: 'test-className' }}/>);
        expect(toJSON(mask)).toMatchSnapshot();
    });
});

