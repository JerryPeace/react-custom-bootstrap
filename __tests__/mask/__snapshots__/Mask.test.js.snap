// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mask render Mask correctly 1`] = `
<Mask
  isLoading={true}
  loaderProps={
    Object {
      "bsSize": "md",
    }
  }
  show={true}
>
  <div
    className="tm-mask"
  >
    <div>
      <Loader
        bsClass="tm-loader"
        bsSize="md"
      >
        <span
          className="loader"
        >
          <span
            className="tm-loader tm-loader-md"
          />
        </span>
      </Loader>
    </div>
  </div>
</Mask>
`;

exports[`Mask render Mask with loaderProps correctly 1`] = `
<Mask
  isLoading={true}
  loaderProps={
    Object {
      "className": "test-className",
    }
  }
  show={true}
>
  <div
    className="tm-mask"
  >
    <div>
      <Loader
        bsClass="tm-loader"
        bsSize="sm"
        className="test-className"
      >
        <span
          className="loader test-className"
        >
          <span
            className="tm-loader tm-loader-sm"
          />
        </span>
      </Loader>
    </div>
  </div>
</Mask>
`;

exports[`Mask render Mask without isLoading correctly 1`] = `
<Mask
  isLoading={false}
  loaderProps={
    Object {
      "bsSize": "md",
    }
  }
  show={true}
>
  <div
    className="tm-mask"
  >
    <div />
  </div>
</Mask>
`;

exports[`Mask render Mask without show correctly 1`] = `
<Mask
  loaderProps={
    Object {
      "bsSize": "md",
    }
  }
  show={false}
/>
`;
